src/main/java/com/gluonhq/substrate/target/AndroidTargetConfiguration.java

```diff
@@ -127,23 +123,32 @@ public boolean packageApp() throws IOException, InterruptedException {
         prepareAndroidResources();
         copyOtherDalvikClasses();
         copySubstrateLibraries();
-        generateSigningConfiguration();
+        String configuration = generateSigningConfiguration();
 
         ProcessRunner assembleDebug = new ProcessRunner(
                             getAndroidProjectPath().resolve("gradlew").toString(),
                             "-p", getAndroidProjectPath().toString(),
-                            "assembleDebug");
+                            "assemble" + configuration);
         assembleDebug.addToEnv("ANDROID_HOME", sdk);
         assembleDebug.addToEnv("JAVA_HOME", projectConfiguration.getGraalPath().toString());
-        return assembleDebug.runProcess("package-task") == 0;
+        if (assembleDebug.runProcess("package-task") != 0) {
+            return false;
+        }
+        Path generatedApk = getAndroidProjectPath().resolve("app").resolve("build")
+                            .resolve("outputs").resolve("apk").resolve(configuration.toLowerCase())
```

Comment:
Use `toLowerCase(Locale.ROOT)`.
Can you also check if `Files.exists(generatedApk)` before copying it to gvm? In case, for any reason, the process ends successfully but the apk file is not found?

Reply:
Should be fixed now